apm-sniffer/apm-sdk-plugin/dbcp-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/dbcp/v2/PoolingSetUrlInterceptor.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.skywalking.apm.plugin.dbcp.v2;
+
+import org.apache.commons.dbcp2.BasicDataSourceMXBean;
+import org.apache.skywalking.apm.agent.core.logging.api.ILog;
+import org.apache.skywalking.apm.agent.core.logging.api.LogManager;
+import org.apache.skywalking.apm.agent.core.meter.MeterFactory;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.plugin.jdbc.connectionurl.parser.URLParser;
+import org.apache.skywalking.apm.plugin.jdbc.trace.ConnectionInfo;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * {@link PoolingSetUrlInterceptor} intercepted the method of DBCP set url and register metric monitor.
+ */
+public class PoolingSetUrlInterceptor implements InstanceMethodsAroundInterceptor {
+    private static final String METER_NAME = "datasource";
+    private static final ILog LOGGER = LogManager.getLogger(PoolingSetUrlInterceptor.class);
+    private static final Map<String, Function<BasicDataSourceMXBean, Supplier<Double>>> METRIC_MAP = new HashMap<String, Function<BasicDataSourceMXBean, Supplier<Double>>>();
```

Comment:
This is fine, considering `BasicDataSourceMXBean` is only an interface. But still, this class metadata will be loaded into the default agent class loader, which could cause unnecessary memory costs. It is better to move it to a constructor? As this interceptor would only be created when a new pool is created, which is a rare case in the runtime, we don't need to be concerned about the performance. 

`TomcatThreadExecutorInterceptor` using `ThreadPoolExecutor` is fine because the ThreadPoolExecutor is loaded by JDK's class loader, then the default agent class loader doesn't need to load again.